modules/monitors/main.tf

```diff
@@ -6,7 +6,7 @@ locals {
 
 # https://registry.terraform.io/providers/DataDog/datadog/latest/docs/resources/monitor
 resource "datadog_monitor" "default" {
-  for_each = local.enabled ? var.datadog_monitors : {}
+  for_each = local.enabled ? { for k, v in var.datadog_monitors : k => v if lookup(v, "enabled", true) } : {}
```

Comment:
did you try it with `module.this.enabled = false`.
In some cases, Tf complains about both side of the `?` operator not having the same types of elements.
If that's an issue, we can do 

```
for_each =  { for k, v in var.datadog_monitors : k => v if local.enabled && lookup(v, "enabled", true) } 
``